src/main/java/com/mycila/xmltool/XMLDocPath.java

```diff
@@ -38,12 +38,14 @@
  */
 final class XMLDocPath {
 
+    private static final XPathFactory XPATH_FACTORY = XPathFactory.newInstance();

```

Comment:
This change would not made it thread safe. You would have to use ThreadLocal to make is thread safe. Just like we do for DateFormat as

private static ThreadLocal<SimpleDateFormat> dateFormat = new ThreadLocal<SimpleDateFormat>() {
        @Override
        protected SimpleDateFormat initialValue() {
            return new SimpleDateFormat(DEF_DATE_TIME_FORMAT);
        }
    };